thrust/thrust/device_ptr.h

```diff
@@ -77,13 +77,7 @@ class device_ptr
   using super_t = thrust::pointer<T, thrust::device_system_tag, thrust::device_reference<T>, thrust::device_ptr<T>>;
 
 public:
-  /*! \brief Construct a null \c device_ptr.
-   *
-   *  \post <tt>get() == nullptr</tt>.
-   */
-  _CCCL_HOST_DEVICE device_ptr()
-      : super_t()
-  {}
+  device_ptr() = default;
```

Comment:
Nitpick, we should also mark them as hidden
```suggestion
  _CCCL_HIDE_FROM_ABI device_ptr() = default;
```